kolibri/core/auth/middleware.py

```diff
@@ -64,3 +66,14 @@ def process_request(self, request):
             "'kolibri.core.auth.middleware.CustomAuthenticationMiddleware'."
         )
         request.user = SimpleLazyObject(lambda: _get_user(request))
+
+    def process_response(self, request, response):
```

Comment:
Might be simpler to add this only in the session endpoint - we are always guaranteed to hit it, and it means we only do this once every four minutes, and not on every request to the server.

Can add this cookie setting here instead: https://github.com/learningequality/kolibri/blob/release-v0.12.x/kolibri/core/auth/api.py#L379

Reply:
Note - my ulterior motive for preferring this is that modifying the session during a request makes Django automatically add `Vary: Cookies` header to any request, which makes it much harder to cache, so if we can restrict these kind of things only to user specific requests, it is preferable!

Reply:
@rtibbles I'll give that a look in the AM tomorrow. I'm a bit confused since I'm using the `HttpResponse.set_cookie()` method for setting the cookie on the rather than setting it on the `session` object using `__setitem__()`.

That point in the code you reference has a session object and @jamalex wants to use a cookie of our own making instead so that it will never expire.

Reply:
While this doesn't modify the session, as @nucleogenesis notes, so it may or may not have an effect on caching, moving this out of a middleware and into the session endpoint would also help by reducing the overhead somewhat -- even though this is a very lightweight middleware (no IO, just CPU), it is still adding a layer to the onion for every single request Django is processing.

Reply:
One update you would need to make is to turn the `get_session` method into a `get_session_response` method, so that you return the `Response` object from the method.

With this, you can use `response.set_cookie` method to set the cookie in the anonymous case.